node/events-to-ga/src/utils.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Throws an error if any of the keys are missing from the object
+ * @param {*} obj
+ * @param {string[]} keys
+ * @throws {Error}
+ */
+export function throwIfMissing(obj, keys) {
+  const missing = [];
+  for (let key of keys) {
+    if (!(key in obj) || !obj[key]) {
+      missing.push(key);
+    }
+  }
+  if (missing.length > 0) {
+    throw new Error(`Missing required fields: ${missing.join(", ")}`);
+  }
+}
+
+/**
+ * returns whether the request headers and appropriate message are ok
+ * @param {*} req
+ * @returns {}
+ */
+export function verifyHeaders(req) {
+  if (req.headers["x-appwrite-user-id"] == "") {
+    throw `x-appwrite-user-id value in req.headers is not there`;
+  }
+  if (req.headers["x-appwrite-trigger"] != "event") {
+    throw `Not triggered by event but by ${req.headers["x-appwrite-trigger"]}`;
+  }
+  if (req.headers["x-appwrite-event"] == "") {
+    throw `x-appwrite-event value in req.headers is not there`;
+  }
+}
+
+/**
+ * returns formatted string by converting appwrite event name to Google Analytics accepted string
+ * @param {string} str
+ * @returns {{event_name:string,wildCardObject:{[key:string]:string}}}
+ */
+export function formatIntoGoogleAnalyticsEvent(str) {
+  const oddElemArray = [];
+  const wildCardArray = [];
+  const splitArray = str.split(".");
+
+  for (let i = 0; i < splitArray.length; i++) {
+    if (i % 2 == 0) oddElemArray.push(splitArray[i]);
+    else
+      wildCardArray.push([
+        `${oldElemArray[oldElemArray.length - 1]}Id`,
+        splitArray[i],
+      ]);
```

Comment:
Please add curly braces here, we prefer to use common syntax so the template is accessible